compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -86,8 +86,10 @@ public void visit(BLangConstant constant) {
         BConstantSymbol tempCurrentConstSymbol = this.currentConstSymbol;
         this.currentConstSymbol = constant.symbol;
         this.currentConstSymbol.value = visitExpr(constant.expr);
-        unresolvedConstants.remove(this.currentConstSymbol);
-        this.currentConstSymbol = tempCurrentConstSymbol;
+        if (this.currentConstSymbol.value != null) {
+            unresolvedConstants.remove(this.currentConstSymbol);
+            this.currentConstSymbol = tempCurrentConstSymbol;
```

Comment:
Shouldn't this be reset irrespective of whether or not the constant was resolved.

Reply:
Yes. I had done a mistake here.